google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -500,9 +501,16 @@ public FirestoreRpc getClient() {
     return firestoreClient;
   }
 
+  /** This method is obsolete. Use {@link #getTotalRequestTimeoutDuration()} instead. */
+  @ObsoleteApi("Use getTotalRequestTimeoutDuration() instead")
   @Override
-  public Duration getTotalRequestTimeout() {
-    return firestoreOptions.getRetrySettings().getTotalTimeout();
+  public org.threeten.bp.Duration getTotalRequestTimeout() {
+    return toThreetenDuration(getTotalRequestTimeoutDuration());
+  }
+
+  @Override
+  public java.time.Duration getTotalRequestTimeoutDuration() {
+    return firestoreOptions.getRetrySettings().getTotalTimeoutDuration();
```

Comment:
nit: Would be nice to get javadocs from firestore for this

Reply:
@ehsannas for visibility - the superclass does not have javadocs on this method